polaris-react/src/components/DataTable/README.md

```diff
@@ -437,6 +440,143 @@ function FullDataTableExample() {
 }
 ```
 
+### Data table with fixed first column
+
+Use when the table contains many columns and it would benefit the merchant to see the first column when scrolling to the right.
+
+```jsx
+function DataTableFixedFirstColumnExample() {
+  const [sortedRows, setSortedRows] = useState(null);
+
+  const initiallySortedRows = [
+    [
+      'Emerald Silk Gown',
+      'Formalwear',
+      "Jill's formal",
+      10,
+      '$875.00',
+      124689,
+      140,
+      '$426.00',
+      '$122,500.00',
+    ],
+    [
+      'Mauve Cashmere Scarf',
+      'Accessories',
+      "Jack's Accessories",
+      253,
+      '$230.00',
+      124533,
+      83,
+      '$620.00',
+      '$19,090.00',
+    ],
+    [
+      'Navy Merino Wool Blazer with khaki chinos and yellow belt',
+      'Ensembles',
+      'Avocado Fashions',
+      23,
+      '$445.00',
+      124518,
+      32,
+      '$353.00',
+      '$14,240.00',
+    ],
+    [
+      'Socks',
+      'Essentials',
+      'Avocado Fashions',
+      465,
+      '$4.00',
+      124518,
+      32,
+      '$3.00',
+      '$140.00',
+    ],
+  ];
+  const rows = sortedRows ? sortedRows : initiallySortedRows;
+
+  const handleSort = useCallback(
+    (index, direction) => setSortedRows(sortRows(rows, index, direction)),
+    [rows],
+  );
+
+  return (
+    <Page title="Sales by product">
+      <Card>
+        <DataTable
+          columnContentTypes={[
+            'text',
+            'text',
+            'text',
+            'numeric',
+            'numeric',
+            'numeric',
+            'numeric',
+            'numeric',
+            'numeric',
+          ]}
+          headings={[
+            'Product',
+            'Category',
+            'Vendor',
+            'Orders',
+            'Price',
+            'SKU Number',
+            'Net quantity',
+            'Shipping',
+            'Net sales',
+          ]}
+          rows={rows}
+          totals={['', '', '', '', '', '', 255, '$1399', '$155,830.00']}
+          sortable={[true, true, true, true, true, true, true, true, true]}
+          defaultSortDirection="descending"
+          initialSortColumnIndex={4}
+          onSort={handleSort}
+          hasFixedFirstColumn
+          truncate
+        />
+      </Card>
+    </Page>
+  );
+
+  function sortRows(rows, index, direction) {
+    return [...rows].sort((rowA, rowB) => {
+      let type;
+      if (rowA[index].length > 1) {
+        type = !isNaN(parseFloat(rowA[index]))
+          ? 'number'
+          : !isNaN(parseFloat(rowA[index].substring(1)))
+          ? 'currency'
+          : 'string';
```

Comment:
For thorough accessibility testing purposes, I wanted all columns to be sortable, so I had to write this function that could sort strings, currency, and bare numbers. I'm sure there's a tidier way to write it, but I think it's worth having that ability to manually test all the columns since we will want that in the Reports tables.